### ISSUE TITLE
EventSourceEvents generates excessive exceptions (and doesn't log events)

### ISSUE THREAD

**Issue by rrector** [dotnet/orleans#6617]
While debugging, noticed a large number of exceptions in the diagnostic tools window:

`'System.ArgumentException' in mscorlib.dll ("EventSource expects the first parameter of the Event method to be of type Guid and to be named "relatedActivityId" when calling WriteEventWithRelatedActivityId.")` with a stack trace that points at `EventSourceEvents`.

Looking at `EventSourceEvents`, all of the implementations look like this:

```
        [Event(1)]
        public void OnTimeout(Guid activityId)
        {
            this.WriteEventWithRelatedActivityId(1, activityId);
        }
```

The [EventSource.WriteEventWithRelatedActivityId documentation](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource.writeeventwithrelatedactivityid?view=netcore-3.1) indicates that there are some requirements for how the event methods need to be written: 

>Your ETW event method calling this function must follow these guidelines:
>
>Specify the first parameter as a Guid named relatedActivityId.
>
>Specify either Send or Receive as the EventAttribute.Opcode property.
>
>Call WriteEventWithRelatedActivityId passing in the event ID, followed by the related ID GUID, followed by all the parameters the event method is passed, in the same order.

In the EventSourceEvents implementation, all of the first parameters are named activityId rather than relatedActivityId, and none of the events specify an EventAttribute.Opcode property.

It's also a little unclear why `message.TraceContext.ActivityId` is being added as a related activity identifier rather than setting it to be the actual ActivityId of the event using `EventSource.SetCurrentThreadActivityId`.

**Comment by rrector** [comment]
A little more info on an easy way to reproduce this (assuming you have Visual Studio).

In Visual Studio, go to View -> Other Windows -> Diagnostic Events.

In the Diagnostic Events window that comes up, click on the settings icon (the gear). In the list of ETW Providers that comes up, add `Microsoft-Orleans-OutsideRuntimeClient`.

Then run any Orleans workflow that sends requests/receives responses. You will then see events in the Diagnostic Events window that look like this:

```
{
  "Timestamp": "2020-06-27T17:10:02.3540307-07:00",
  "ProviderName": "Microsoft-Orleans-OutsideRuntimeClient",
  "Id": 0,
  "Message": "EventSourceException while processing event \"SendRequest\"",
  "ProcessId": 25296,
  "Level": "Always",
  "Keywords": "0xFFFFFFFFFFFFFFFF",
  "EventName": "EventSourceMessage",
  "ActivityID": "00004160-0001-0000-d062-0000ffdcd7b5",
  "RelatedActivityID": null,
  "Payload": {
    "message": "EventSourceException while processing event \"SendRequest\""
  }
}
{
  "Timestamp": "2020-06-27T17:10:02.3880753-07:00",
  "ProviderName": "Microsoft-Orleans-OutsideRuntimeClient",
  "Id": 0,
  "Message": "EventSourceException while processing event \"SendRequest\": System.ArgumentException:EventSource expects the first parameter of the Event method to be of type Guid and to be named \"relatedActivityId\" when calling WriteEventWithRelatedActivityId.",
  "ProcessId": 25296,
  "Level": "Always",
  "Keywords": "0xFFFFFFFFFFFFFFFF",
  "EventName": "EventSourceMessage",
  "ActivityID": "00004160-0001-0000-d062-0000ffdcd7b5",
  "RelatedActivityID": null,
  "Payload": {
    "message": "EventSourceException while processing event \"SendRequest\": System.ArgumentException:EventSource expects the first parameter of the Event method to be of type Guid and to be named \"relatedActivityId\" when calling WriteEventWithRelatedActivityId."
  }
}
```

**Comment by sergeybykov** [comment]
Resolved via #6626.
